iwyu.cc

```diff
@@ -2146,6 +2146,61 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
         }
         return true;
       }
+      case TypeTrait::BTT_IsConvertible:
+      case TypeTrait::BTT_IsConvertibleTo:
+      case TypeTrait::BTT_IsNothrowConvertible: {
+        // No conversion to/from void or to a function type.
+        if (lhs_type->isVoidType() || rhs_type->isVoidType())
+          return true;
+        if (rhs_type->isFunctionType())
+          return true;
+        // Only non-reference-related types can involve user-defined conversions
+        // (C++17 [dcl.init.ref]).
+        if (GetCanonicalType(RemoveReference(lhs_type)) ==
+            GetCanonicalType((RemoveReference(rhs_type)))) {
+          return true;
+        }
+        const bool rhs_can_bind_to_temp = RefCanBindToTemp(rhs_type);
+        // If the rhs is a reference that can bind to its class temporary
+        // object, the ctors of the class are to be considered.
+        if (rhs_can_bind_to_temp)
+          ReportTypeUse(CurrentLoc(), rhs_type, DerefKind::RemoveRefs);
+        // Pointer and pointer-to-member conversions should be taken into
+        // account if the rhs type is a pointer or pointer-to-member type
+        // directly, or a reference which can bind to a temporary pointer
+        // object.
+        if (!rhs_type->isReferenceType() || RefCanBindToTemp(rhs_type)) {
```

Comment:
Can use `rhs_can_bind_to_temp` instead of invoking `RefCanBindToTemp` again

Reply:
Fixed, thanks!